optuna/integration/chainermn.py

```diff
@@ -124,7 +125,19 @@ def optimize(
                 if not has_next_trial:
                     break
                 trial = Trial(self.delegate, trial_id)
-                func(trial, self.comm)
+                try:
+                    # We assume that if a node raises an exception, all other nodes does it.
+                    func(trial, self.comm)
+                except TrialPruned:
+                    pass
```

Comment:
In the case of pruned trials, it is useful to offer an option to automatically re-run the failed trials so that the full trial count is achieved. `i.e. chainermn_study.optimize(objective, n_trials=25, rerun_pruned=True)` such that `len(study.trials) == n_trials == n_trials + (rerun_trials - pruned_trials)`

Reply:
Thank you for your comment. I think the proposed feature is not ChainerMN specific and out of the scope of this PR. Therefore, it seems appropriate to create an issue for discussing the feature.

Reply:
Thanks, it is more appropriate as a separate issue. I have now done so.

Reply:
Thank you. Because Optuna's developers are on new year's holidays, we will check it next week.